crates/task-impls/src/consensus.rs

```diff
@@ -1199,12 +1210,22 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                 "Sending proposal for view {:?} \n {:?}",
                 leaf.view_number, ""
             );
-            self.event_stream
-                .publish(HotShotEvent::QuorumProposalSend(
-                    message,
-                    self.public_key.clone(),
-                ))
-                .await;
+
+            futures::join! {
+                self.event_stream
+                    .publish(HotShotEvent::QuorumProposalSend(
+                        message.clone(),
+                        self.public_key.clone(),
+                    )),
+                self.api
+                    .send_event(Event {
+                        view_number: self.cur_view,
+                        event: EventType::QuorumProposal {
+                            proposal: message,
+                            sender: self.public_key.clone(),
+                        },
+                    })
+            };
```

Comment:
Personally, I feel it is cleaner to just send the event when we receive the proposal.  If we send the external event here, the application will receive two events, one from here and one when it receives the proposal internally.

Reply:
Sorry, I'm not sure I follow. How does the application receive duplicate events in this case?

Reply:
The application will receive 2 `EventType::QuorumProposal` external events: 1 from when HotShot sends the proposal to the other nodes (which is the case above), and 1 from when HotShot receives the proposal (from itself).  So if a node is the proposer for a particular view, it will end up sending 2 of these events to the application.

Reply:
Oh, nevermind. The sender receives the proposal as well, duh.

Reply:
@elliedavidson right, sorry, GitHub didn't yet show me your comment when I was responding. Thank you for the explanation